pkg/providers/imagefamily/image.go

```diff
@@ -40,41 +40,36 @@ type Provider struct {
 	kubernetesInterface    kubernetes.Interface
 	imageCache             *cache.Cache
 	imageVersionsClient    CommunityGalleryImageVersionsAPI
+	subscription           string
+	NodeImageVersions      NodeImageVersionsAPI
 }
 
 const (
 	kubernetesVersionCacheKey = "kubernetesVersion"
 
+	// if imageVersion is equal to an "", then we will get the latest version
+	autoUpgradeMode = ""
```

Comment:
Since we are looking to remove `imageVersion` I think leaving this out for now would be best.

Reply:
Oh, missed this is a const. Makes sense for clarity. However, should still be getting removed.